eng/pipelines/common/ui-tests-steps.yml

```diff
@@ -0,0 +1,104 @@
+parameters:
+  platform: '' # [ android, ios, windows, catalyst ]
+  path: '' # path to csproj
+  device: '' # the xharness device to use
+  cakeArgs: '' # additional cake args
+  app: '' #path to app to test
+  version: '' #the iOS version'
+  provisionatorChannel: 'latest'
+  agentPoolAccessToken: ''
+  configuration : "Release"
+
+steps:
+
+  - template: provision.yml
+    parameters:
+      ${{ if eq(parameters.platform, 'windows') }}:
+        platform: windows
+      ${{ if ne(parameters.platform, 'windows') }}:
+        platform: macos
+      skipXcode: false
+      provisionatorChannel: ${{ parameters.provisionatorChannel }}
+
+  - task: PowerShell@2
+    condition: ne('${{ parameters.platform }}' , 'windows')
+    inputs:
+      targetType: 'inline'
+      script: |
+        defaults write -g NSAutomaticCapitalizationEnabled -bool false
+        defaults write -g NSAutomaticTextCompletionEnabled -bool false
+        defaults write -g NSAutomaticSpellingCorrectionEnabled -bool false
+    displayName: "Modify defaults"
+    continueOnError: true
+
+  - task: NodeTool@0
+    inputs:
+      versionSpec: "18.13.0"
+    displayName: "Install node"
+
+  - task: PowerShell@2
+    inputs:
+      targetType: 'inline'
+      script: |
+        node -v
+        npm install -g appium@next
+        appium -v
+        $x = appium driver list --installed --json  | ConvertFrom-Json
+        if(!$x.windows)
+        {
+            appium driver install --source=npm appium-windows-driver
+        }
+        if(!$x.uiautomator2)
+        {
+            appium driver install uiautomator2
+        }
+        if(!$x.xcuitest)
+        {
+            appium driver install xcuitest
+        }
+        if(!$x.mac2)
+        {
+            appium driver install mac2
+        }
+    displayName: "Install Appium"
+    continueOnError: true
+
+  - pwsh: ./build.ps1 --target=dotnet --configuration="${{ parameters.configuration }}" --verbosity=diagnostic
+    displayName: 'Install .NET'
+    retryCountOnTaskFailure: 3
+    env:
+      DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
+      PRIVATE_BUILD: $(PrivateBuild)
+
+  - pwsh: echo "##vso[task.prependpath]$(DotNet.Dir)"
+    displayName: 'Add .NET to PATH'
+
+  - pwsh: ./build.ps1 --target=dotnet-buildtasks --configuration="${{ parameters.configuration }}"
+    displayName: 'Build the MSBuild Tasks'
+
+  - pwsh: ./build.ps1 --target=dotnet-samples --configuration="${{ parameters.configuration }}" --${{ parameters.platform }} --verbosity=diagnostic 
```

Comment:
@pjcollins  for example here i want to build the samples to use the app, but i might just need to create a target just to build my test app for the tests

Reply:
I moved the sample building tests into the integration test project in PR https://github.com/dotnet/maui/pull/14499. You could potentially call that to build the sample projects, though maybe those tests are now moot if these tests will build _and_ run the samples?